internal/scheduler/preempting_queue_scheduler.go

```diff
@@ -852,12 +853,20 @@ func (evi *Evictor) Evict(ctx *armadacontext.Context, nodeDbTxn *memdb.Txn) (*Ev
 		}
 
 		for _, job := range evictedJobs {
-			// Create a scheduling context for when re-scheduling this job.
-			// Mark as evicted and add a node selector to ensure the job is re-scheduled onto the node it was evicted from.
+			// Create a scheduling context for the attempt to re-schedule the job, and:
+			// 1. Mark the job as evicted. This ensures total scheduled resources is calculated correctly.
+			// 2. Add a node selector ensuring the job can only be re-scheduled onto the node it was evicted from.
+			// 3. Add tolerations for all taints on the node. This to ensure that:
+			//    - Adding taints to a node doesn't cause jobs already running on the node to be preempted.
+			//    - Jobs scheduled as away jobs have the necessary tolerations to be re-scheduled.
+			// TODO(albin): We can remove the checkOnlyDynamicRequirements flag in the nodeDb now that we've added the tolerations.
 			jctx := schedulercontext.JobSchedulingContextFromJob(evi.priorityClasses, job)
 			jctx.IsEvicted = true
 			jctx.AddNodeSelector(schedulerconfig.NodeIdLabel, node.Id)
 			evictedJctxsByJobId[job.GetId()] = jctx
+			for _, taint := range node.Taints {
```

Comment:
This was the second issue. We didn't previously pass through tolerations added when scheduling a job as an away-job.